workspaces/adventure-pack/goodies/python3/src/flatten/__init__.py

```diff
@@ -0,0 +1,9 @@
+from typing import Generator, Any
+
+
+def flatten(self: list[Any]) -> Generator[Any, None, None]:
+    for i in self:
+        if isinstance(i, list):
+            yield from flatten(i)
```

Comment:
Nice, I like it.

One thing to consider: this flattens one level of the input however I believe in many libraries `flatten` recursively flattens nested lists as well, or at least allow you to configure how many levels of nesting to flatten. I'm cool with starting with this, but it would be nice to allow for deeper flattening in the future.

Reply:
I believe this code does recursively flatten nested lists to a single layer. 
Would we still want to configure how many layers it flattens?

Reply:
Ah ok, I missed that for some reason. Yeah, I think it might be nice to configure. JavaScript supports this (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) as does Ruby (https://apidock.com/ruby/Array/flatten).

Reply:
I created #387.